src/App.tsx

```diff
@@ -1,27 +1,51 @@
-import React from 'react';
-import './App.scss';
+import React, { useState } from 'react';
+
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+import { Good } from './types/Good';
+import { Callback } from './types/Callback';
+
+import './App.scss';
+
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
+
+  const getGoods = async (callback: Callback) => {
+    const goods = await callback();
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+    setVisibleGoods(goods);
+  };
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => getGoods(getAll)}
```

Comment:
```suggestion
        onClick={makeGetGoods(getAll)}
```
consider to make fuction factory for all buttons and just call in JSX
```
const makeGetGoods = (callBack: Callback) => () => getGoods(callback);
```